lib/services/metadata_provider.dart

```diff
@@ -30,7 +32,7 @@ class MetadataRequest {
   }
```

Comment:
You should update the == override to match the hashCode.

Reply:
Oh, didn't see that. Any reason why we don't just use the `hashCode` for the comparison?

Reply:
Possible hashcode collisions maybe? I don’t know, I’m mostly just following the docs here.

Reply:
I'm glad at least one of us is 👀